packages/flutter_tools/lib/src/commands/migrate_status.dart

```diff
@@ -0,0 +1,163 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/platform.dart';
+import '../base/terminal.dart';
+import '../migrate/migrate_manifest.dart';
+import '../migrate/migrate_utils.dart';
+import '../project.dart';
+import '../runner/flutter_command.dart';
+import 'migrate.dart';
+
+/// Flutter migrate subcommand to check the migration status of the project.
+class MigrateStatusCommand extends FlutterCommand {
+  MigrateStatusCommand({
+    bool verbose = false,
+    required this.logger,
+    required this.fileSystem,
+    required Platform platform,
+    required ProcessManager processManager,
+  }) : _verbose = verbose,
+       migrateUtils = MigrateUtils(
+         logger: logger,
+         fileSystem: fileSystem,
+         platform: platform,
+         processManager: processManager,
+       ) {
+    requiresPubspecYaml();
+    argParser.addOption(
+      'working-directory',
+      help: 'Specifies the custom migration working directory used to stage and edit proposed changes. '
+            'This path can be absolute or relative to the flutter project root.',
+      valueHelp: 'path',
+    );
+    argParser.addOption(
+      'project-directory',
+      help: 'The root directory of the flutter project.',
```

Comment:
for simplicity, should it only assume the current directory is the project directory?

Reply:
This tool is also intended to be used programmatically, where this option helps immensely.

Reply:
It looks like if this is not provided we default to the current working directory, can we document this?